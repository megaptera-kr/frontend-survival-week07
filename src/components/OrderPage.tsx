import { Link } from 'react-router-dom';

export default function OrderPage() {
  return (
    <>
      <h1>이대로 주문하시겠어요?</h1>
      <Link to="/complete">결제</Link>
    </>
  );
}
