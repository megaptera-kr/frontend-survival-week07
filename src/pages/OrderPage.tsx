import { useNavigate } from 'react-router-dom';

import useCartStore from '../hooks/useCartStore';
import PATH from '../constants/path';

export default function OrderPage() {
  const navigate = useNavigate();

  const [, store] = useCartStore();

  const handleOrder = async () => {
    await store.clearItem();
    navigate(PATH.orderCompleted);
  };

  return (
    <section>
      <h1>이대로 주문하시겠어요?</h1>
      <button type="button" onClick={handleOrder}>결제</button>
    </section>
  );
}
