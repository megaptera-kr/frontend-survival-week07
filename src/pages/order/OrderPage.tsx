import Typo from '../../components/Typo';
import Cart from '../../containers/Cart';
import RestaurantsSearch from '../../containers/RestaurantsSearch';
import Restaurants from '../../containers/Resturants';

function OrderPage() {
  return (
    <div data-testid="OrderPage" className="order-wrapper">
      <section className="header">
        <RestaurantsSearch />
        <p>
          <Typo level={2}>점심 바구니</Typo>
        </p>
      </section>
      <section className="content">
        <Restaurants />
        <Cart />
      </section>
    </div>
  );
}

export default OrderPage;
