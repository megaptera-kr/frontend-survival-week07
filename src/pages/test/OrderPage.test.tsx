import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Route, useLocation } from 'react-router-dom';

import OrderPage from '../OrderPage';
import { withRouter } from '../../tests/helper';
import PATH from '../../constants/path';

const context = describe;

describe('Order Page', () => {
  function OrderCompleted() {
    const { pathname } = useLocation();
    return <pre>{JSON.stringify(pathname)}</pre>;
  }

  function renderOrderPage() {
    return render(withRouter(
      <>
        <Route path="/" element={<OrderPage />} />
        <Route path={PATH.orderCompleted} element={<OrderCompleted />} />
      </>,
    ));
  }

  it('render correctly', () => {
    const { asFragment } = renderOrderPage();

    expect(asFragment()).toMatchSnapshot();
  });

  context('when payment button is clicked', () => {
    it('order button is disabled', async () => {
      renderOrderPage();

      const paymentBtn = screen.getByRole('button', { name: '결제' });
      await userEvent.click(paymentBtn);

      await waitFor(() => expect(screen.getByText(`"${PATH.orderCompleted}"`)).toBeInTheDocument());
    });
  });
});
