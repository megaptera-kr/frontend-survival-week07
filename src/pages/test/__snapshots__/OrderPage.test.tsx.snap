// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Page render correctly 1`] = `
<DocumentFragment>
  <section>
    <h1>
      이대로 주문하시겠어요?
    </h1>
    <button
      type="button"
    >
      결제
    </button>
  </section>
</DocumentFragment>
`;
